worker/runfmusite/runFMUSite.py

```diff
@@ -0,0 +1,226 @@
+   Inputs:  variable 
+   Returns: print statement on the terminal
+   '''
+   print(')))))) Hey i am checking '+ str(var) + '((((((: ', var)
+
+
+
+####################   Entry for Main Program   #####################
+############################################################################
+
+
+
+try:
+    s3 = boto3.resource('s3', region_name='us-east-1', endpoint_url=os.environ['S3_URL'])
+
+    #Initiate Mongo Database
+    mongo_client = MongoClient(os.environ['MONGO_URL'])
+    mongodb = mongo_client[os.environ['MONGO_DB_NAME']]
+    recs = mongodb.recs
+
+    #get user inputs
+    site_ref = sys.argv[1]
+    real_time_flag = sys.argv[2]
+    time_scale = int(sys.argv[3])
+    user_start_Datetime = parse(sys.argv[4])
+    user_end_Datetime = parse(sys.argv[5])
+
+    #build the path for zipped-file, fmu, json
+    sim_path = '/simulate'
+    directory = os.path.join(sim_path, site_ref)
+    tar_name = "%s.tar.gz" % site_ref
+    key = "parsed/%s" % tar_name
+    tarpath = os.path.join(directory, tar_name)
+    fmupath = os.path.join(directory, 'model.fmu')
+    tagpath = os.path.join(directory, 'tags.json')
+
+        
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    
+    #download the tar file and tag file
+    bucket = s3.Bucket('alfalfa')
+    bucket.download_file(key, tarpath)
+    bucket.download_file(key, tagpath)
+    
+    tar = tarfile.open(tarpath)
+    tar.extractall(sim_path)
+    tar.close()
+    
+    #get the tagging id, disp, etc......
+    tag_data = get_tag_data(tagpath)
+
+    recs.update_one({"_id": site_ref}, {"$set": {"rec.simStatus": "s:Running"}}, False)
+
+    
+
+    # Load fmu
+    config = {
+        'fmupath'  : fmupath,                
+        'step'     : 60
+    }
+
+        
+    (dis_and_id, tagid_and_inputs, tagid_and_outputs) = \
+            create_DisToID_dictionary(tagpath)
+ 
+
+    #initiate the testcase
+    tc = testcase.TestCase(config) 
+
+    #setup the fake inputs
+    u={}
+    for each_input in tc.get_inputs():
+        if each_input !='time':
+            u[each_input]=1.0   
+ 
+    #run the FMU simulation
+    kstep=0 
+    #while tc.start_time <= 1000000000000:
+    for i in range(10):
+        time.sleep(5)
+        tc.advance(u)
+        output = tc.get_results()
+        
+        u_output = output['u']
+        y_output = output['y']
+        for key in u_output.keys():
+            value_u = u_output[key]
+            
+            if key!='time':
+                input_id = tagid_and_inputs[key]
+                                
+                cur_value = 3.33
+                recs.update_one( {"_id": input_id }, {"$set": {"rec.curVal":"n:%s" %cur_value, "rec.curStatus":"s:ok","rec.cur": "m:" }} )
+
+        for key in y_output.keys():
+            value_y = y_output[key]
+            
+            if key!='time': 
+                output_id = tagid_and_outputs[key]
+                             
+                cur_value =273.15
```

Comment:
output handling looks good. we just need to use value_y instead of the hard coded cur_value

Reply:
yes. I hard-coded the y to make sure the values were sent to the database and shown in the front-end. Updated results are pushed back to repo.